jsapp/js/app.jsx

```diff
@@ -99,28 +118,29 @@ class App extends React.Component {
                 <BigModal params={this.state.pageState.modal} />
               )}
 
-              {!this.isFormBuilder() && (
-                <React.Fragment>
+              {this.shouldDisplayMainLayoutElements() && (
+                <>
                   <MainHeader assetUid={assetid} />
                   <Drawer />
-                </React.Fragment>
+                </>
```

Comment:
Because of some pre-existing memory leaks, this introduces some errors when the `<MainHeader/>` and some other components are unmounted. (Previously, those components wouldn't unmount when entering the NLP view.)

![image](https://github.com/kobotoolbox/kpi/assets/7819986/2b0d45a9-4f7c-413c-a5b5-d36cbc0e5438)

Ultimately, it's a bit of an upstream problem, so I'll put together a PR to fix those - it's not blocking this review, which works fine otherwise in my testing.